validation.js

```diff
@@ -0,0 +1,148 @@
+console.log('validation.js loaded check');
+document.addEventListener('DOMContentLoaded', function () {
+    const emailLogin = document.getElementById('email-login');
+    const passwordLogin = document.getElementById('password-login');
+    const loginButton = document.getElementById('login-btn');
+
+    const emailSignup = document.getElementById('email-signup');
+    const nicknameSignup = document.getElementById('nickname');
+    const passwordSignup = document.getElementById('password-signup');
+    const passwordConfirmationSignup = document.getElementById('password-confirmation');
+    const signupButton = document.getElementById('signup-btn');
+
+    function validateEmail(email) {
+        const regex = /^[^\s@]+@[^\s@]+.[^\s@]+$/;
+
+        return regex.test(email);
+    }
+
+    function validatePassword(password) {
+        return password.length >= 8;
+    }
+
+    function toggleButtonState() {
+        const isValidLogin =
+            emailLogin?.value &&
+            passwordLogin?.value &&
+            validateEmail(emailLogin.value) &&
+            validatePassword(passwordLogin.value);
+        if (loginButton) loginButton.disabled = !isValidLogin;
+
+        const isValidSignup =
+            emailSignup?.value &&
+            nicknameSignup?.value &&
+            passwordSignup?.value &&
+            passwordConfirmationSignup?.value &&
+            validateEmail(emailSignup.value) &&
+            validatePassword(passwordSignup.value) &&
+            passwordSignup.value === passwordConfirmationSignup.value;
+        if (signupButton) signupButton.disabled = !isValidSignup;
+    }
+
+    function showError(input, message) {
+        input.classList.add('error');
+        const errorMessage = input.closest('.form-group').querySelector('.error-message');
```

Comment:
💊 제안
자바스크립트에서 이런 변수들은 호이스팅됩니다. 따라서 가능한 변수, 함수선언문들은 해당 코드블록 최상단에 두시는 것이 좋습니다.

```suggestion
        const errorMessage = input.closest('.form-group').querySelector('.error-message');
        input.classList.add('error');   
```

[mdn Hoisting](https://developer.mozilla.org/ko/docs/Glossary/Hoisting)